wsmaster/che-core-api-machine-shared/src/main/java/org/eclipse/che/api/machine/shared/dto/MachineSourceDto.java

```diff
@@ -30,10 +31,33 @@
     MachineSourceDto withType(String type);
 
     @Override
-    @FactoryParameter(obligation = MANDATORY)
+    @FactoryParameter(obligation = OPTIONAL)
```

Comment:
In that case it will be possible to save factory with neither location nor content. Does it seem OK to you @benoitf ?